### DISCUSSION TITLE
Setting innerHTML in next <head>

### DISCUSSION THREAD

**Discussion by simplenotezy** [vercel/next.js#24749] [Help]
How can I set innerHTML using next's <Head> component? https://nextjs.org/docs/api-reference/next/head

I have tried with:

          <Head
            dangerouslySetInnerHTML={{
              __html: seo.fullHead,
            }}
          />

But that does not inject anything. I have also tried this: https://github.com/vercel/next.js/issues/3904#issuecomment-447716569 without any luck.

The reason is, that I use an GraphQL extensions that supports returning out all the HTML needed for the head section for a given page, so that I don't need to individually specify each field manually. The `seo.fullHead` is a string containing page title, og:title, noindex and many other meta tags needed in the head section.

**Comment by bobarros** [comment]
It's not allowed. 

You can do something like:

```html
<Head>
        <script dangerouslySetInnerHTML={{ __html: 'something' }} />
</Head>
```

You can't do what you're trying to:

```html
 <Head dangerouslySetInnerHTML={{__html: 'some code'}} />
```

**Reply by simplenotezy** [reply]
Thanks for your reply. The inner html I wish to set contantains <title>, <meta>, etc tags, and cannot be wrapped in script. Is there a way to set innerHtml on an element, e.g. `<>` so that an outer element is not necessary?

**Comment by bobarros** [comment]
No, there's not. 
As you might know, dangerouslySetInnerHTML has this name because it's meant to not be used.

_______
_title, meta or any other elements (e.g. script) need to be contained as direct children of the Head element, or wrapped into maximum one level of <React.Fragment> or arrays—otherwise the tags won't be correctly picked up on client-side navigations._
_______

Why are you avoiding using it as regular props?

```jsx
const YourPage = ({ seo.fullHead }) => {
return (
<>
 <Head>  
<title>{seo.fullHead.title}</title>
 ...and so
</Head>
<Body>something</Body>
</>
)

export async function getStaticProps() {
//...getting seo.fullHead
  return {
    seo.fullHead as props
  }
}

```
